example-plugins/inj-tyfam-plugin/src/Plugin/InjTyFam.hs

```diff
@@ -0,0 +1,243 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Plugin.InjTyFam (
+  plugin,
+  tcPlugin,
+  ) where
+
+import           Control.Monad (guard)
+import           Data.Maybe (fromMaybe)
+
+import qualified Coercion
+import qualified CoreSyn
+import qualified Plugins
+import           TcEvidence (EvTerm)
+import qualified TcPluginM
+import qualified TcRnMonad
+import qualified TcRnTypes
+import           TcRnTypes (Ct, Xi)
+import           TcType (TcType)
+import qualified TyCoRep
+import           TyCoRep (Coercion)
+import qualified TyCon
+import           TyCon (TyCon)
+import qualified Type
+import qualified Unify
+import           VarEnv (VarEnv)
+import qualified VarEnv
+
+-- | Just 'tcPlugin', with 'Plugins.purePlugin' set
+
+plugin :: Plugins.Plugin
+plugin = Plugins.defaultPlugin
+    {
+      Plugins.pluginRecompile = Plugins.purePlugin
+    ,
+      Plugins.tcPlugin = \_opts -> pure tcPlugin
+    }
+
+-- | A typechecker plugin that decomposes Given equalities headed by
+-- an injective type family,
+-- <https://downloads.haskell.org/ghc/8.6.5/docs/html/users_guide/glasgow_exts.html#injective-type-families>
```

Comment:
use "latest" version of the guide
https://downloads.haskell.org/ghc/latest/docs/html/users_guide/glasgow_exts.html#injective-type-families

Reply:
I went with 8.10.1. I'm hesitant to use `latest` because I've seen link paths change -- I'd rather send them to an old page than a 404.